enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,398 @@
+
+## Open Questions [optional]
+
+* When should the client-based prototype be deprecated and removed?
+* What is the bar for the delivery of USC via a new image in the payload? (see note in [Update Status Controller][this-usc])
+* What is the best architecture for the future system where the USC only aggregates and summarizes information (possibly provided in the form of Update Insights) from external components that want to contribute update-related information?
+
+## Test Plan
+
+* Tests for the Status API will be added to the `openshift/origin` test suite, covering both update and non-update paths. In non-update paths, we will test that the API correctly reports the cluster as not updating. In update paths, the presence of the expected status insights will be monitored, and their properties checked. We can consider tightening the test suite to fail in the presence of health insights because these signal undesirable conditions during the update.
+* USC code will reside in the CVO repository, which has CI jobs for TechPreview installs and both update directions (into the change and out of the change).
+* The client `oc adm upgrade status` code that interprets the API to human-readable output will receive similar [integration example-based tests][oc-adm-upgrade-status-examples] as the client-based implementation.
+* All code will be appropriately unit-tested.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+N/A - the `UpdateStatus` feature gate is already Tech Preview
+
+### Tech Preview -> GA
+
+* The API exists and is marked as v1.
+* The USC is running in the cluster and maintains the `UpdateStatus` resource in standalone mode.
+* The appropriate API support level is decided.
+* The `UpdateStatus` capability exists, is enabled by default, and allows admins to opt out through standard mechanisms (pinning a version capability set to one without it or `None`).
+* A clear plan to achieve HyperShift support is in place.
+* The `oc adm upgrade status` consumes the Status API by default and has at least feature parity with the client-based prototype.
+* Meets TRT criteria: e2e tests exist in `openshift/origin`, and a result data corpus proves the feature works and does not lower platform stability.
+
+### Removing a deprecated feature
+
+N/A
```

Comment:
Might be worth talking about the removal of the tech-preview client-side logic here?